components/emoji_picker/emoji_picker.vue

```diff
@@ -0,0 +1,240 @@
+        :search-no-results-label="searchNoResultsLabel"
+        :recently-used-emojis="recentlyUsedEmojis"
+        :selected-tabset="selectedTabset"
+        @scroll-into-tab="updateScrollIntoTab"
+        @is-scrolling-with-scroll-to="updateIsScrollingWithScrollTo"
+        @emoji-data="updateEmojiData"
+        @selected-emoji="emits('selected-emoji', $event)"
+      />
+    </div>
+    <div class="d-emoji-picker--footer">
+      <emoji-description :emoji="emojiData" />
+      <emoji-skin-selector
+        :is-hovering="!!emojiData"
+        :skin-selector-button-tooltip-label="skinSelectorButtonTooltipLabel"
+        :skin-tone="skinTone"
+        @skin-tone="emits('skin-tone', $event)"
+      />
+    </div>
+  </div>
+</template>
+
+<script setup>
+import EmojiSearch from './modules/emoji_search.vue';
+import EmojiTabset from './modules/emoji_tabset.vue';
+import EmojiSelector from './modules/emoji_selector.vue';
+import EmojiSkinSelector from './modules/emoji_skin_selector.vue';
+import EmojiDescription from './modules/emoji_description.vue';
+import { computed, onBeforeUnmount, ref } from 'vue';
+
+const props = defineProps({
+  /**
+   * The array with recently used  emoji object
+   * This list is necessary to fill the recently used tab
+   * @type {Array}
+   * @default []
+   * @example
+   * <dt-emoji-picker :recentlyUsedEmojis="[emojiObject, emojiObject]" />
+   */
+  // TODO try to simplify this to achieve an array of unicode characters and not an entire emoji data object
+  recentlyUsedEmojis: {
+    type: Array,
+    default: () => ([]),
+  },
+
+  /**
+   * The placeholder text for the search input
+   * @type {String}
+   * @required
+   * @example
+   * <dt-emoji-picker :searchPlaceholderLabel="'Search...'" />
+   */
+  searchPlaceholderLabel: {
+    type: String,
+    required: true,
+  },
+
+  /**
+   * The label for the search results tab
+   * @type {String}
+   * @required
+   * @example
+   * <dt-emoji-picker :searchResultsLabel="'Search results'" />
+   */
+  searchResultsLabel: {
+    type: String,
+    required: true,
+  },
+
+  /**
+   * The label for the search no results
+   * @type {String}
+   * @required
+   * @example
+   * <dt-emoji-picker :searchNoResultsLabel="'No results'" />
+   */
+  searchNoResultsLabel: {
+    type: String,
+    required: true,
+  },
+
+  /**
+   * The list of tabsets to show, it is necessary to be updated with the correct language
+   * It must respect the provided order.
+   * @type {Array}
+   * @default ['Most recently used', 'People', 'Nature', 'Food', 'Activity', 'Travel', 'Objects', 'Symbols', 'Flags']
+   * @example
+   * <dt-emoji-picker :tabSetLabels="['Most recently used', 'People', 'Nature', 'Food', 'Activity', 'Travel', 'Objects', 'Symbols', 'Flags']" />
+   */
+  tabSetLabels: {
+    type: Array,
+    default: () => ['Most recently used', 'Smileys and people', 'Nature', 'Food', 'Activity', 'Travel', 'Objects', 'Symbols', 'Flags'],
```

Comment:
I updated it to required prop and I provided it from the story.
It will be documented in the storybook.

Now if you dont provide a tabSetLabels it will throw an error and not work.